dids/diddht/internal/bep44/bep44.go

```diff
@@ -36,10 +36,11 @@ type bep44Message struct {
 	v []byte
 }
 
-type signer func(payload []byte) ([]byte, error)
+// Signer is a function that signs a given payload and returns the signature.
+type Signer func(payload []byte) ([]byte, error)
 
-// newSignedBEP44Message creates a new signed BEP44 message with the given DNS payload.
-func newSignedBEP44Message(dnsPayload []byte, seq int64, publicKeyBytes []byte, signer signer) (*bep44Message, error) {
+// NewSignedBEP44Message bemcodes the payload, signes it with the signer and creates a new BEP44 message with the given sequence number, public key.
+func NewSignedBEP44Message(dnsPayload []byte, seq int64, publicKeyBytes []byte, signer Signer) (*Message, error) {
```

Comment:
we can shorten the name of this factory function to just `NewMessage`.  `bep44.NewSignedBEP44Message` feels a bit redundant. I also didn't see an `Unsigned` variant so i think we can do away with `Signed`. Lmk if you feel having `Signed` in the name provides context you believe is helpful at call sites

Reply:
agreed. also there's a signer in the params, so it should be obvious.